src/fortran/client.F90

```diff
@@ -136,24 +136,24 @@ module smartredis_client
 
 !> Initializes a new instance of a SmartRedis client
 subroutine initialize( client, cluster )
```

Comment:
This should also be `self`

Reply:
This is the constructor, so I thought it merited special treatment. Do you disagree?

Reply:
Could you elaborate as to why you think it would warrant a deviation? The main one that I can think of is that it'd be easier to grep for the initialization routines for a given type. From my point of view is that these initialize routines are still a type-bound procedure and so it seems more consistent to me that all such procedures should use `self`. 

Maybe to satisfy both we could rename the subroutine name and signature to be
```
 subroutine initialize_client(self, cluster)
```

and then in the definition of the type:

```
type, public :: client_type

[...]

contains

procedure, public :: initialize => initialize_client

end type client_type
```

Thoughts?

Reply:
Works for me!

Reply:
Works for me. I'll do something similar for dataset

Reply:
Changes complete, review requested